torchgeo/datasets/chesapeake.py

```diff
@@ -584,11 +583,6 @@ def __init__(
         lc_colors = lc_colors[:, :3] / 255
         self._lc_cmap = ListedColormap(lc_colors)
 
-        nlcd_colors = np.zeros((max(self.nlcd_cmap.keys()) + 1, 4))
-        nlcd_colors[list(self.nlcd_cmap.keys())] = list(self.nlcd_cmap.values())
```

Comment:
The tests are failing because the cmap isn't in the format expected by matplotlib. It might be simplest to use:
```python
nlcd_colors = np.zeros((max(NLCD.cmap.keys()) + 1, 4))
nlcd_colors[list(NLCD.cmap.keys())] = list(NLCD.cmap.values())
```
instead. Then we don't need to set `self.cmap` since there may be 2 different cmaps for the same dataset. You can also delete `nlcd_cmap` above to avoid duplication.